ui/client/components/graph/Graph.js

```diff
@@ -72,8 +78,29 @@ class Graph extends React.Component {
 		this.processGraphPaper.setDimensions(area.offsetWidth, area.offsetHeight)
 	}
 
+	copyNode(event) {
+		if (!NodeUtils.noChosenNode(this.props.nodeToDisplay) && this.props.allModalsClosed) {
+			navigator.clipboard.writeText(JSON.stringify(this.props.nodeToDisplay))
+		}
+	}
+
+	pasteNode(event) {
+		if (!this.props.allModalsClosed) {
+			return
+		}
+		const clipboardText = (event.clipboardData || window.clipboardData).getData('text');
+		const node = JsonUtils.tryParseOrNull(clipboardText)
+		if (!this.props.capabilities.write || !NodeUtils.isNode(node) || NodeUtils.nodeIsGroup(node)) {
```

Comment:
maybe move this.addNode into if - so there are fewer returns?
I also wonder if it's possible/easy to generate new id if the current one is already used - e.g. when copy/pasting to same diagram?

Reply:
Done. You're right, for now new id is always generated even if it's unique.